### PR TITLE
add unit test for usecase, repository, PhotoMapper

### PR DESCRIPTION
**Pull request by severianfw**

### CODE REVIEW

File: app/src/test/java/com/severianfw/picto/usecase/SearchPhotoUseCaseTest.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        MockitoAnnotations.openMocks(this)
+    }
+
+    @After
+    fun tearDown() {
+        Mockito.verifyNoMoreInteractions(photoRepository)
+    }
+
+    @Test
+    fun testInvoke() {
```
Comment (WennYanto) [inline app/src/test/java/com/severianfw/picto/usecase/SearchPhotoUseCaseTest.kt]: Let's make method something like this

```
fun `when invoke then result should be valid`()
```

File: app/src/test/java/com/severianfw/picto/utils/PhotoMapperTest.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import com.severianfw.picto.data.remote.ProfileImage
+import com.severianfw.picto.data.remote.User
+import com.severianfw.picto.domain.model.PhotoItemModel
+import org.junit.Assert
+import org.junit.Test
+
+class PhotoMapperTest {
+
+    @Test
+    fun mapToPhotoItemModelTest() {
```
Comment (WennYanto) [inline app/src/test/java/com/severianfw/picto/utils/PhotoMapperTest.kt]: Missing unit test when `PhotoResponse()` where all fields are null

File: app/src/test/java/com/severianfw/picto/usecase/GetPhotoUseCaseTest.kt [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import com.severianfw.picto.utils.PhotoMapper
+import io.reactivex.rxjava3.core.Single
+import org.junit.After
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.mockito.InjectMocks
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.Mockito.doNothing
```
Comment (WennYanto) [inline app/src/test/java/com/severianfw/picto/usecase/GetPhotoUseCaseTest.kt]: Let's be consistent. Either `Mockito.when` or `when` , `Mockito.doNothing` or `doNothing`

File: app/src/test/java/com/severianfw/picto/usecase/GetPhotoUseCaseTest.kt [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+        // When
+        val result = getPhotoUseCaseImpl.invoke(dummyPage, dummyIsInitial).blockingGet()
+
+        // Then
+        Assert.assertEquals(result, emptyList<PhotoItemModel>())
+        Mockito.verify(internetConnectionListener).isInternetAvailable()
+    }
+
+
```
Comment (WennYanto) [inline app/src/test/java/com/severianfw/picto/usecase/GetPhotoUseCaseTest.kt]: remove empty lines

File: app/src/test/java/com/severianfw/picto/repository/PhotoRepositoryTest.kt [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        // Then
+        Assert.assertEquals(result, listOf(dummyPhotoResponse))
+        Mockito.verify(photoDao).insertPhotos(listOf(dummyPhotoItemModel))
+        Mockito.verify(apiService).getPhotos(
+            dummyClientId,
+            dummyPerPage,
+            dummyPage
+        )
+    }
+
```
Comment (WennYanto) [inline app/src/test/java/com/severianfw/picto/repository/PhotoRepositoryTest.kt]: remove lines

**Review by WennYanto — APPROVED** [review]
